File: celsius_conversion/ocaml_celsius_conversion/celsius_conversion.ml
(* comment *)

let toFahrenheit(x) = x *. (9.0 /. 5.0) +. 32.0;;
let toCelsius(x) = (x -. 32.0) *. (5.0 /. 9.0);;

let main () =
  if  Array.length Sys.argv <> 1 then
    begin
      Printf.printf "%s <noargs>\n" Sys.argv.(0);
      exit 1
    end
  else
    begin
      Printf.printf "toFahrenheit(%d)=<%f>\n" 21 (toFahrenheit(21.0));
      Printf.printf "toCelsius(%d)=<%f>\n" 70 (toCelsius(70.0))
    end

let _ = main ()

File: celsius_conversion/c_celsius_conversion/celsius_conversion.c
#include <stdio.h>
#include <stdlib.h>

double toFahrenheit ( double );
double toCelsius ( double );

int main( int argc, char *argv[] ) {
  if( argc != 1 ) {
    fprintf( stderr, "Usage: %s <noargs>\n", argv[0]);
    exit( 1 );
  }

  printf("toFahrenheit(%f)=<%f>\n", 21.0, toFahrenheit(21.0));
  printf("toCelsius(%f)=<%f>\n", 70.0, toCelsius(70.0));
  return 0;
}

double toFahrenheit ( double x ) {
  return  x * (9.0/5.0) + 32.0;
}

double toCelsius ( double x ) {
  return (x - 32.0) * (5.0/9.0);
}
